### PR TITLE
SDIT-1396 add a new appearance to an existing court case

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 1]
```diff
@@ -144,24 +142,13 @@ class SentencingService(
             startTime = courtCase.courtEvents[0].nextEventStartTime!!,
             courtEventType = courtCase.courtEvents[0].courtEventType,
             eventStatus = lookupEventStatusType(EventStatus.SCHEDULED), // TODO confirm scheduled is always the status for next appearance
-            prison = lookupEstablishment(request.courtAppearance.nextCourtId),
+            prison = lookupEstablishment(request.courtAppearance.nextCourtId!!), // if next event, we must have a specified court
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:145]: could default the next court to the current court if not provided

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt [hunk 2]
```diff
@@ -1061,67 +1066,289 @@ class SentencingResourceIntTest : IntegrationTestBase() {
...
-    ) =
-      OffenderChargeRequest(
-        offenceCode = offenceCode,
-        statuteCode = statuteCode,
-        offencesCount = offencesCount,
-        offenceDate = offenceDate,
-        offenceEndDate = offenceEndDate,
-        resultCode1 = resultCode1,
-        mostSeriousFlag = mostSeriousFlag,
-      )
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt:1125]: moving to be shared by multiple tests

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 3]
```diff
@@ -314,6 +314,89 @@ class SentencingResource(private val sentencingService: SentencingService) {
     request: CreateCourtCaseRequest,
   ): CreateCourtCaseResponse =
     sentencingService.createCourtCase(offenderNo, request)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_SENTENCING')")
+  @PostMapping("/prisoners/{offenderNo}/sentencing/court-cases/{caseId}")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:319]: minor - but I think a URL of `/prisoners/{offenderNo}/sentencing/court-cases/{caseId}/court-appearance` probably reads better

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 4]
```diff
@@ -314,6 +314,89 @@ class SentencingResource(private val sentencingService: SentencingService) {
...
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = CourtAppearanceRequest::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Created Court case",
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:335]: minor copy n paste

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 5]
```diff
@@ -314,6 +314,89 @@ class SentencingResource(private val sentencingService: SentencingService) {
...
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun createCourtAppearance(
+    @Schema(description = "Booking Id", example = "12345", required = true)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:390]: type booking id v offender no

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt (hunk 3 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:319]: agreed 👍

**Pull request merged**